### ISSUE TITLE
Error Prone + Maven not writing patch file(s)

### ISSUE THREAD

**Issue by maximeblais** [google/error-prone#2086]
### Description of the problem / feature request:

I am using Error Prone in combination with Maven in a multi-module project with the following configuration of the `maven-compiler-plugin`:
```xml
<plugin>
    <groupId>org.apache.maven.plugins</groupId>
    <artifactId>maven-compiler-plugin</artifactId>
    <configuration>
        <compilerArgs>
            <arg>-XDcompilePolicy=simple</arg>
            <arg>-Xplugin:ErrorProne -XepPatchChecks:MissingOverride,DefaultCharset,DeadException -XepPatchLocation:${project.build.sourceDirectory}</arg>
        </compilerArgs>
        <annotationProcessorPaths>
            <path>
                <groupId>com.google.errorprone</groupId>
                <artifactId>error_prone_core</artifactId>
                <version>2.4.0</version>
            </path>
        </annotationProcessorPaths>
    </configuration>
</plugin>
```

My expectation was that this would create multiple patch files that could be subsequently applied to sources files as needed. However, even though files are being reported as written in the console, they definitely aren't.

Example message from the console:
```
Changes were written to /Users/XXX/<path-to-project>/<module-name>/src/main/java/error-prone.patch. Please inspect the file and apply with: patch -p0 -u -i error-prone.patch
```

I don't know if this is relevant but I am on macOS Catalina 10.15.7

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

- Setup a maven project with a parent POM and at least 1 submodule and edit the maven-compiler-plugin configuration to match the above.
- Add a source file containing code that fails at least one of the specified checks.
- Compile the project with maven.
- Observe that no files are written.

### What version of Error Prone are you using?

2.4.0

###  Have you found anything relevant by searching the web?

I've search through closed issues here and went over a couple of StackOverflow posts but didn't find anything relevant to my issue.

**Comment by cushon** [comment]
> * Setup a maven project with a parent POM and at least 1 submodule and edit the maven-compiler-plugin configuration to match the above.
> * Add a source file containing code that fails at least one of the specified checks.
> * Compile the project with maven.
> * Observe that no files are written.

I cannot reproduce, following those steps in the following example results in a patch file being written. Can you provide a complete example that demonstrates the problem?

If you use a different path instead of `${project.build.sourceDirectory` for `-XepPatchLocation:` (e.g. a temporary directory outside of the source root), does that make a difference?

```
for f in $(find . -type f); do echo "=== $f"; cat $f; done
=== ./lib/src/main/java/com/foo/Foo.java
package com.foo;

final class Foo {
  void f() {
    new Exception();
  }
}
=== ./lib/pom.xml
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
<modelVersion>4.0.0</modelVersion>

<artifactId>lib</artifactId>

<parent>
  <groupId>org.test</groupId>
  <artifactId>super-pom</artifactId>
  <version>1.0.0-SNAPSHOT</version>
</parent>

</project>
=== ./pom.xml
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
<modelVersion>4.0.0</modelVersion>

<groupId>org.test</groupId>
<artifactId>super-pom</artifactId>
<packaging>pom</packaging>
<version>1.0.0-SNAPSHOT</version>

<modules>
  <module>lib</module>
</modules>

<build>
  <plugins>
    <plugin>
      <groupId>org.apache.maven.plugins</groupId>
      <artifactId>maven-compiler-plugin</artifactId>
      <version>3.8.0</version>
      <configuration>
        <source>8</source>
        <target>8</target>
        <compilerArgs>
          <arg>-XDcompilePolicy=simple</arg>
          <arg>-Xplugin:ErrorProne -XepPatchChecks:MissingOverride,DefaultCharset,DeadException -XepPatchLocation:${project.build.sourceDirectory}</arg>
        </compilerArgs>
        <annotationProcessorPaths>
          <path>
            <groupId>com.google.errorprone</groupId>
            <artifactId>error_prone_core</artifactId>
            <version>2.4.0</version>
          </path>
        </annotationProcessorPaths>
      </configuration>
    </plugin>
  </plugins>
</build>
</project>
```

```
$ mvn clean package
...
Changes were written to ./lib/src/main/java/error-prone.patch. Please inspect the file and apply with: patch -p0 -u -i error-prone.patch
```

```diff
$ cat lib/src/main/java/error-prone.patch 
--- com/foo/Foo.java
+++ com/foo/Foo.java
@@ -3,5 +3,5 @@
 final class Foo {
   void f() {
-    new Exception();
+    throw new Exception();
   }
 }
```

**Comment by maximeblais** [comment]
hey @cushon thanks for taking the time to provide me with a working example. I've tried it on my side and it worked perfectly.

I then went back to my original project to compare the configurations, and they were the same. I re-ran mvn and it worked this time, so I'm not sure what I did wrong when I first opened this issue...

Anyways, thank you again ! I will close the issue and re-open another one if need be.
